api/lang/diagnostic.go

```diff
@@ -1,32 +1,57 @@
 package lang
 
 import (
-	"github.com/getgauge/gauge/gauge"
 	"github.com/getgauge/gauge/parser"
 	"github.com/getgauge/gauge/util"
 	"github.com/sourcegraph/go-langserver/pkg/lsp"
 )
 
 func createDiagnostics(uri string) []lsp.Diagnostic {
-	diagnostics := make([]lsp.Diagnostic, 0)
 	file := util.ConvertURItoFilePath(uri)
-	_, res := new(parser.SpecParser).Parse(getContent(uri), gauge.NewConceptDictionary(), file)
-	if len(res.ParseErrors) > 0 {
-		for _, err := range res.ParseErrors {
-			diagnostics = append(diagnostics, createDiagnostic(err, uri))
+	var res *parser.ParseResult
+	if util.IsConcept(file) {
+		res = validateConcept(uri, file)
+	} else {
+		_, res = new(parser.SpecParser).Parse(getContent(uri), provider.GetConceptDictionary(), file)
+	}
+	return createDiagnosticsFrom(res, uri)
+}
+
+func validateConcept(uri string, file string) *parser.ParseResult {
+	dictionary := provider.GetConceptDictionary()
+	for _, cpt := range dictionary.ConceptsMap {
+		if cpt.FileName == file {
+			delete(dictionary.ConceptsMap, cpt.ConceptStep.Value)
```

Comment:
Concepts cache and Concept dictionary in specInfoGatherer may go out of sync. Also, all changes should be done within info Gatherer to ensure all caches maintained are upto date. Modifications outside it should not be done.